tests/ui/const_is_empty.rs

```diff
@@ -0,0 +1,14 @@
+#![warn(clippy::const_is_empty)]
+
+fn main() {
```

Comment:
Could you add these test cases as well? They should all pass:

```rs
fn str_from_arg(var: &str) {
    var.is_empty();
}

fn update_str() {
    let mut value = "duck";
    value = "penguin";

    let _ = value.is_empty();
}

fn macros() {
    // Content from Macro
    let file = include_str!("const_is_empty.rs");
    let _ = file.is_empty();

    let var = env!("PATH");
    let _ = var.is_empty();
}

fn conditional_value() {
    let value;

    if true {
        value = "hey";
    } else {
        value = "hej";
    }

    let _ = value.is_empty()
}
```

Reply:
I've added them as the fourth commit so that you can check what is detected and what is not.

Reply:
Concerning the `macros` test, do you mean that you would want them not to lint?